packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -122,25 +148,57 @@ export class GitClientService {
     throw new InvalidPullRequestMode();
   }
 
-  private async createBranchIfNotExists(args: GetBranchArgs): Promise<Branch> {
+  private async restoreAmplicationBranchIfNotExists(
+    args: CreateBranchIfNotExistsArgs
+  ): Promise<Branch> {
+    const { branchName, owner, repositoryName, gitClient } = args;
     const branch = await this.provider.getBranch(args);
     if (!branch) {
       const { defaultBranch } = await this.provider.getRepository(args);
-      const { sha } = await this.provider.getFirstCommitOnBranch({
-        ...args,
-        branchName: defaultBranch,
+      const firstCommitOnDefaultBranch =
+        await this.provider.getFirstCommitOnBranch({
+          owner,
+          repositoryName,
+          branchName: defaultBranch,
+        });
+      const branch = await this.provider.createBranch({
+        owner,
+        branchName,
+        repositoryName,
+        pointingSha: firstCommitOnDefaultBranch.sha,
       });
-      const branch = this.provider.createBranch({ ...args, pointingSha: sha });
       const amplicationCommits = await this.provider.getCurrentUserCommitList({
-        ...args,
+        owner,
+        repositoryName,
         branchName: defaultBranch,
       });
-
+      await this.cherryPickCommits({
+        commits: amplicationCommits,
+        gitClient,
+        branchName,
+        firstCommitOnDefaultBranch,
+      });
       return branch;
     }
     return branch;
   }
 
+  private async cherryPickCommits(args: CherryPickCommitsArgs) {
```

Comment:
this is a private function, I do not see a reason to create Args object - it just make the code less readable